### PR TITLE
Expand palette id filter

### PR DESCRIPTION
**Pull request by harrisonpim**
adds sierra catalogue ids to the set of queryable ids, and a coherent script for generating a new `identifiers.pkl` file, on which `identifiers.py` is now based

### CODE REVIEW

File: apis/palette_api/app/api.py [hunk 1]
```diff
@@ -18,27 +17,21 @@
     docs_url='/docs',
     redoc_url='/redoc'
 )
-# CORS
```
Comment (jamieparkinson) [inline apis/palette_api/app/api.py]: How come this has been removed?

File: apis/palette_api/app/api.py [hunk 2]
```diff
@@ -18,27 +17,21 @@
...
-)
 
 # create API endpoints
 
 
 @app.get('/works/{catalogue_id}')
 def palette_similarity_by_catalogue_id(catalogue_id: str, n: int = 10):
-    catalogue_id = catalogue_id or np.random.choice(catalogue_ids)
-    if catalogue_id not in catalogue_ids:
+    catalogue_id = catalogue_id or np.random.choice(valid_catalogue_ids)
```
Comment (jamieparkinson) [inline apis/palette_api/app/api.py]: fwiw I don't think the random choice will ever be selected as `/works/` and `/works` will 404 first, but that's fine imo as a random result for a non-specified ID is not especially RESTful

File: apis/palette_api/generate_identifiers.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
+import json
```
Comment (jamieparkinson) [inline apis/palette_api/generate_identifiers.py]: Maybe worth putting this file in a `/scripts` or something dir? not sure

**Comment by jamieparkinson** [comment]
Only actual request for change or explanation here is the CORS middleware comment

**Comment by jamieparkinson** [comment]
Resolves https://github.com/wellcometrust/platform/issues/4078

File: apis/palette_api/app/api.py (hunk 1 above)
Comment (harrisonpim) [reply apis/palette_api/app/api.py]: no idea when/how that was removed... it's back now

File: apis/palette_api/generate_identifiers.py (hunk 3 above)
Comment (harrisonpim) [reply apis/palette_api/generate_identifiers.py]: not intending to leave this here long term - i think it's worth puzzling out the problem more thoroughly in `data-pipedream` discussions

**Pull request merged**